src/miral/basic_window_manager.cpp

```diff
@@ -2016,6 +2016,36 @@ auto miral::BasicWindowManager::place_new_surface(WindowSpecification parameters
 
         if (parameters.top_left().value().y < display_area.top_left.y)
             parameters.top_left() = Point{parameters.top_left().value().x, display_area.top_left.y};
+
+        if (parameters.state() == mir_window_state_restored)
+        {
+            auto const offset{48};
+            std::array const positions {
+                parameters.top_left().value(),
+                parameters.top_left().value() + Displacement( offset,  offset),
+                parameters.top_left().value() + Displacement(-offset,  offset),
+                parameters.top_left().value() + Displacement( offset, -offset),
+                parameters.top_left().value() + Displacement(-offset, -offset)};
+
+            for (auto const& position : positions)
+            {
+                auto const window{window_at(position)};
+                auto const window_type{info_for(window).type()};
+                auto const window_state{info_for(window).state()};
+
+                if (!window ||
+                    window_state != mir_window_state_restored ||
+                    (window_type != mir_window_type_normal &&
+                     window_type != mir_window_type_decoration &&
+                     window_type != mir_window_type_freestyle) ||
+                     window.top_left() != position)
```

Comment:
The logic here seems wrong.

1. `!window` cannot happen: a logic error would have been thrown by `info_for`
2. We don't want to exclude restored windows
3. We don't want to exclude normal and freestyle windows

Reply:
Sorry, I misread the condition (which itself suggests it could be clearer).